crates/languages/src/python.rs

```diff
@@ -182,57 +182,94 @@ async fn get_cached_server_binary(
-        },
-        TaskTemplate {
-            label: format!("run '{}'", VariableName::File.template_value()),
-            command: "python3".to_owned(),
-            args: vec![VariableName::File.template_value()],
-            ..TaskTemplate::default()
-        },
-        TaskTemplate {
-            label: format!("unittest '{}'", VariableName::File.template_value()),
-            command: "python3".to_owned(),
-            args: vec![
-                "-m".to_owned(),
-                "unittest".to_owned(),
-                VariableName::File.template_value(),
-            ],
-            ..TaskTemplate::default()
-        },
-        TaskTemplate {
-            label: "unittest $ZED_SYMBOL TestCase".to_owned(),
-            command: "python3".to_owned(),
-            args: vec![
-                "-m".to_owned(),
-                "unittest".to_owned(),
-                VariableName::RelativeFile.template_value(),
-                "-k".to_owned(),
-                "$ZED_SYMBOL".to_owned(),
-            ],
-            tags: vec!["python-unittest-class".to_owned()],
-            ..TaskTemplate::default()
-        },
-        TaskTemplate {
-            label: "unittest $ZED_SYMBOL function".to_owned(),
-            command: "python3".to_owned(),
-            args: vec![
-                "-m".to_owned(),
-                "unittest".to_owned(),
-                VariableName::RelativeFile.template_value(),
-                "-k".to_owned(),
-                "$ZED_SYMBOL".to_owned(),
-            ],
-            tags: vec!["python-unittest-function".to_owned()],
-            ..TaskTemplate::default()
-        },
-    ]))
+pub(crate) struct PythonContextProvider;
+
+const PYTHON_UNITTEST_TARGET_TASK_VARIABLE: VariableName =
+    VariableName::Custom(Cow::Borrowed("PYTHON_UNITTEST_TARGET"));
+
+impl ContextProvider for PythonContextProvider {
+    fn build_context(
+        &self,
+        _variables: &task::TaskVariables,
+        _location: &project::Location,
+        _cx: &mut gpui::AppContext,
+    ) -> Result<task::TaskVariables> {
+        let _python_module_name = python_module_name_from_relative_path(
+            _variables.get(&VariableName::RelativeFile).unwrap_or(""),
+        );
+        let _unittest_class_name =
+            _variables.get(&VariableName::Custom(Cow::Borrowed("_unittest_class_name")));
+        let _unittest_method_name = _variables.get(&VariableName::Custom(Cow::Borrowed(
+            "_unittest_method_name",
+        )));
+
+        let _unittest_target_str = match (_unittest_class_name, _unittest_method_name) {
+            (Some(class_name), Some(method_name)) => {
+                format!("{}.{}.{}", _python_module_name, class_name, method_name)
+            }
+            (Some(class_name), None) => format!("{}.{}", _python_module_name, class_name),
+            (None, None) => format!("{}", _python_module_name),
+            (None, Some(_)) => unreachable!("Method name without class name is not possible"),
```

Comment:
we probably shouldn't crash the whole app, and just return an `Ok(TaskVariables::default())` instead.